### PR TITLE
[stable/postgresql] Allow setting the mount path of initdb scripts

### PR DESCRIPTION
**Pull request by mwalser**
#### What this PR does / why we need it:
Some PostgreSQL images such as TimescaleDB are shipping with initdb scripts in /docker-entrypoint-initdb.d/. Using the same directory as mountPath results in these scripts being shadowed.

This change makes it possible to specify a different mount path (for example /docker-entrypoint-initdb.d/helm/) so that these scripts can coexist.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Review by juan131 — COMMENTED** [review]
Hi @mwalser 

Thanks so much for the PR! 

What are the reasons to allow modifying this **mountPath** for initdb scripts? Are you using the Bitnami image? Please note this image expects the initdb scripts at that specific path.

**Comment by mwalser** [comment]
Hi @juan131 

Thank you for the fast reply.

The reason adding this is that currently the path of the ConfigMap mount is fixed to /docker-entrypoint-initdb.d/.
This wouldn't be a problem, but the TimescaleDB docker image (based on bitnami/postgresql) already contains some initialization scripts: https://github.com/timescale/timescaledb-docker/tree/master/docker-entrypoint-initdb.d

When we now use the initdb feature of the helm chart only the initdb scripts are visible in /docker-entrypoint-initdb.d/. The preexisting files (000_install_timescaledb.sh, 001_reenable_auth.sh and 002_timescaledb_tune.sh) are inaccessible.

This pull request adds the ability to specify a custom mount path, e.g. /docker-entrypoint-initdb.d/helm/. This enables a coexistence of the already existing files and those from the helm chart.

**Comment by juan131** [comment]
Thanks for the detailed explanation. It makes total sense.

> This pull request adds the ability to specify a custom mount path, e.g. /docker-entrypoint-initdb.d/helm/. This enables a coexistence of the already existing files and those from the helm chart.

Just a question. How are you supposed to run these new scripts during the initialization? I mean, the current logic (the one at Bitnami PostgreSQL image) won't detect your custom scripts (it'll only detect the ones included in the TimescaleDB docker image) and therefore, you need to find a way to execute them

**Comment by mwalser** [comment]
Hi @juan131 

I'm not sure I understand your question.

When the initdb scripts are mounted in `/docker-entrypoint-initdb.d/some-custom-directory/` they are going to be found by this script https://github.com/bitnami/bitnami-docker-postgresql/blob/master/11/centos-7/rootfs/libpostgresql.sh#L580

There is no maxdepth option specified in the find command so it will decend into the custom subdirectory.

The directory layout would look like this:
```
/docker-entrypoint-initdb.d/000_install_timescaledb.sh
/docker-entrypoint-initdb.d/001_reenable_auth.sh
/docker-entrypoint-initdb.d/002_timescaledb_tune.sh
/docker-entrypoint-initdb.d/some-custom-directory/some-initdb-script.sql
/docker-entrypoint-initdb.d/some-custom-directory/other-initdb-script.sql
```

File: stable/postgresql/values.yaml [hunk 1]
```diff
@@ -179,6 +179,9 @@ postgresqlDataDir: /bitnami/postgresql/data
 ## NOTE: This will override initdbScripts
 # initdbScriptsConfigMap:
 
+## The path the initdb scripts will be mounted at
+initdbScriptsMountPath: /docker-entrypoint-initdb.d/
+
```
Comment (juan131) [inline stable/postgresql/values.yaml:184]: Could you do the same change in the **values-production.yaml**?

**Review by juan131 — CHANGES_REQUESTED** [review]

**Comment by juan131** [comment]
> There is no maxdepth option specified in the find command so it will decend into the custom subdirectory.

You're right! Thanks so much again for the explanation.

Please rebase your branch from master to avoid the conflicting issues.

**Comment by zanhsieh** [comment]
@mwalser 
Would you mind to resolve conflicting files please?
stable/postgresql/Chart.yaml
stable/postgresql/README.md
stable/postgresql/templates/statefulset.yaml

**Comment by carrodher** [comment]
Sorry, I just saw the stale bot notification.
Yes, you should fix the existing conflicts in the following files:
```
stable/postgresql/Chart.yaml
stable/postgresql/README.md
stable/postgresql/templates/statefulset.yaml
```
In the same way, as @juan131 suggested, can you add the same change in the _values-production.yaml_?

**Comment by mwalser** [comment]
Sorry, for the late answer. It was a bit of a stressful time. I'm not sure anymore the solution I proposed is the right way to go. While it might solve my specific problem it feels very hacky.

I thought about changing this merge request to always put the initdb scripts into a subfolder. But while this might work with the bitnami based images it will not on the official postgres docker image. So this is no way forward.

The next thought I had was to abandon the idea of letting the user specify the initdb script path via `initdbScriptsMountPath` and to just introduce a boolean `initdbScriptsUseSubfolder` (which would default to false). IMHO this feels slightly less hacky.

Do you have any opinions on this?

**Pull request closed without merging**